src/Traits/EntityControlTrait.php

```diff
@@ -157,20 +158,31 @@ public function create(array $data): Model
 
     public function insert(array $data): bool
     {
-        $timestamps = [];
+        $defaultTimestamps = [];
 
         if ($this->model->timestamps) {
             $now = now();
 
-            $timestamps = [
+            $defaultTimestamps = [
                 $this->model::CREATED_AT => $now,
                 $this->model::UPDATED_AT => $now
             ];
         }
 
-        $data = array_map(function ($item) use ($timestamps) {
+        $data = array_map(function ($item) use ($defaultTimestamps) {
             $fillableFields = Arr::only($item, $this->model->getFillable());
 
+            $timestamps = array_merge($defaultTimestamps, Arr::only($fillableFields, [
+                $this->model::CREATED_AT,
+                $this->model::UPDATED_AT
+            ]));
+
+            array_walk($timestamps, function (&$timestamp) {
+                $timestamp = $timestamp instanceof Carbon
+                    ? $timestamp
+                    : Carbon::parse($timestamp);
+            });
```

Comment:
as discussed, no need to map data into Carbon instance, developer should fill timestamps with the correct data type in case he fill timestamps manually